src/multi.c

```diff
@@ -268,23 +255,6 @@ void execCommand(client *c) {
     c->cmd = orig_cmd;
     discardTransaction(c);
 
-    /* Make sure the EXEC command will be propagated as well if MULTI
-     * was already propagated. */
-    if (server.propagate_in_transaction) {
-        int is_master = server.masterhost == NULL;
-        server.dirty++;
-        /* If inside the MULTI/EXEC block this instance was suddenly
-         * switched from master to slave (using the SLAVEOF command), the
-         * initial MULTI was propagated into the replication backlog, but the
-         * rest was not. We need to make sure to at least terminate the
-         * backlog with the final EXEC. */
-        if (server.repl_backlog && was_master && !is_master) {
-            char *execcmd = "*1\r\n$4\r\nEXEC\r\n";
-            feedReplicationBuffer(execcmd,strlen(execcmd));
-        }
```

Comment:
this block is irrelevant because we never propagate anything before EXEC is completely done

Reply:
right

we didn't, i just added one